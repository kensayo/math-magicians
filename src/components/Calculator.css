.calculator {
  display: grid;
  font-weight: bold;
  grid-template:
    "res res res res" 60px
    "ac sym per div" 60px
    "7 8 9 mul" 60px
    "4 5 6 min" 60px
    "1 2 3 plus" 60px
    "zero zero dot eq" 60px / 90px 90px 90px 90px;
  text-align: center;
  width: fit-content;
}

h1 {
  margin: 0;
  padding: 0;
}

.result {
  grid-area: res;
  display: flex;
  flex-direction: column;
  align-items: end;
  justify-content: end;
  padding: 0 20px 10px 0;
  background-color: lightgray;
  border: 1px solid darkgray;
}

.operator {
  background-color: orange;
}

.operator:hover {
  background-color: lightsalmon !important;
  color: white !important;
}

.zero {
  grid-area: zero;
}

.item {
  border: 1px solid lightgray;
  display: flex;
  align-items: center;
  justify-content: center;
}

.item:hover {
  color: white;
  background-color: darkgray;
}

.item:active {
  background-color: darkgreen;
}

button {
  border: none;
  background: none;
  height: 100%;
  width: 100%;
  cursor: pointer;
}
