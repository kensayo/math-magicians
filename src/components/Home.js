import React from 'react';

const Home = () => (
  <article className="p-5">
    <h1>Welcome to our page!</h1>
    <br />
    <p>
      Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan
      id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec
      sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan
      tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin
      eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue
      leo eget malesuada.
    </p>

    <p>
      Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan
      id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec
      sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan
      tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin
      eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue
      leo eget malesuada.
    </p>

  </article>
);

export default Home;
