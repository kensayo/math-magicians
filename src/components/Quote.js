import { React, useEffect, useState } from 'react';

function Quote() {
  const [error, setError] = useState(null);
  const [isLoaded, setIsLoaded] = useState(false);
  const [items, setItems] = useState('');

  useEffect(() => {
    fetch('https://api.quotable.io/random?minLength=150')
      .then((res) => res.json())
      .then(
        (result) => {
          setIsLoaded(true);
          setItems(result);
        },
        (error) => {
          setIsLoaded(true);
          setError(error);
        },
      );
  }, []);

  if (error) {
    return (
      <div className="d-flex justify-content-center p-5 m-5">
        <p className="w-50 fs-5">
          <strong>Error: </strong>
          {error.message}
        </p>
      </div>
    );
  } if (!isLoaded) {
    return (
      <div className="d-flex justify-content-center p-5 m-5">
        <p className="w-50 fs-5 fw-boldtext-align-center">Loading...</p>
      </div>
    );
  }
  return (
    <div className="d-flex justify-content-center p-5 m-5">
      <p className="w-50 fs-5">
        {items.content}
        {' '}
        -
        {' '}
        <strong>{items.author}</strong>
      </p>

    </div>
  );
}

export default Quote;
