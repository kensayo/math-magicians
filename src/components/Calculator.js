import './Calculator.css';
import { useState, React } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [data, setObj] = useState({ total: '0', next: null, operation: null });
  const { total, next, operation } = data;

  const update = (input) => {
    setObj(calculate(data, input));
  };

  return (
    <div className="container">
      <div className="result">
        <span id="result">
          {total}
          {operation}
          {next}
        </span>
      </div>
      <div className="item"><button type="button" value="AC" onClick={() => update('AC')}>AC</button></div>
      <div className="item"><button type="button" value="+/-" onClick={() => update('+/-')}>+/-</button></div>
      <div className="item"><button type="button" value="%" onClick={() => update('%')}>%</button></div>
      <div className="item operator"><button type="button" value="รท" onClick={() => update('รท')}>รท</button></div>
      <div className="item"><button type="button" text="7" value="7" onClick={() => update('7')}>7</button></div>
      <div className="item"><button type="button" value="8" onClick={() => update('8')}>8</button></div>
      <div className="item"><button type="button" value="9" onClick={() => update('9')}>9</button></div>
      <div className="item operator"><button type="button" value="x" onClick={() => update('x')}>x</button></div>
      <div className="item"><button type="button" value="4" onClick={() => update('4')}>4</button></div>
      <div className="item"><button type="button" value="5" onClick={() => update('5')}>5</button></div>
      <div className="item"><button type="button" value="6" onClick={() => update('6')}>6</button></div>
      <div className="item operator"><button type="button" value="-" onClick={() => update('-')}>-</button></div>
      <div className="item"><button type="button" value="1" onClick={() => update('1')}>1</button></div>
      <div className="item"><button type="button" value="2" onClick={() => update('2')}>2</button></div>
      <div className="item"><button type="button" value="3" onClick={() => update('3')}>3</button></div>
      <div className="item operator"><button type="button" value="+" onClick={() => update('+')}>+</button></div>
      <div className="item zero"><button type="button" value="0" onClick={() => update('0')}>0</button></div>
      <div className="item"><button type="button" value="." onClick={() => update('.')}>.</button></div>
      <div className="item operator"><button type="button" value="=" onClick={() => update('=')}>=</button></div>
    </div>
  );
};

export default Calculator;
