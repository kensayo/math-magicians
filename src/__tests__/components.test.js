import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Quote from '../components/Quote';
import Home from '../components/Home';
import Calculator from '../components/Calculator';

describe('Testing Components', () => {
    test('Home.js', () => {
      const tree = render(<Home/>);
      expect(tree).toMatchSnapshot();
    });
  
    test('Calculator.js', () => {
      const tree = render(<Calculator />);
      expect(tree).toMatchSnapshot();
      fireEvent.click(screen.getByText('8'));
      expect(screen.getByTestId('result')).toHaveTextContent('8');
      fireEvent.click(screen.getByText('+'));
      fireEvent.click(screen.getByText('1'));
      fireEvent.click(screen.getByText('='));
      expect(screen.getByTestId('result')).toHaveTextContent('9');
    });
  
    test('Quote.js', () => {
      const tree = render(<Quote/>);
      expect(tree).toMatchSnapshot();
    });
  });