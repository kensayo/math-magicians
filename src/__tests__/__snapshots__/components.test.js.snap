// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components Calculator.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex w-100 justify-content-evenly"
      >
        <div
          class="mt-5 w-50 px-5"
        >
          <h1>
            Let's do some Math's
          </h1>
        </div>
        <div
          class="d-flex justify-content-center w-50"
        >
          <div
            class="calculator mt-5"
          >
            <div
              class="result"
            >
              <span
                data-testid="result"
                id="result"
              >
                0
              </span>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="AC"
              >
                AC
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="+/-"
              >
                +/-
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="%"
              >
                %
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="รท"
              >
                รท
              </button>
            </div>
            <div
              class="item"
            >
              <button
                text="7"
                type="button"
                value="7"
              >
                7
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="8"
              >
                8
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="9"
              >
                9
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="x"
              >
                x
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="4"
              >
                4
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="5"
              >
                5
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="6"
              >
                6
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="-"
              >
                -
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="1"
              >
                1
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="2"
              >
                2
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="3"
              >
                3
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="+"
              >
                +
              </button>
            </div>
            <div
              class="item zero"
            >
              <button
                type="button"
                value="0"
              >
                0
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="."
              >
                .
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="="
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex w-100 justify-content-evenly"
    >
      <div
        class="mt-5 w-50 px-5"
      >
        <h1>
          Let's do some Math's
        </h1>
      </div>
      <div
        class="d-flex justify-content-center w-50"
      >
        <div
          class="calculator mt-5"
        >
          <div
            class="result"
          >
            <span
              data-testid="result"
              id="result"
            >
              0
            </span>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="AC"
            >
              AC
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="+/-"
            >
              +/-
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="%"
            >
              %
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="รท"
            >
              รท
            </button>
          </div>
          <div
            class="item"
          >
            <button
              text="7"
              type="button"
              value="7"
            >
              7
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="8"
            >
              8
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="9"
            >
              9
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="x"
            >
              x
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="4"
            >
              4
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="5"
            >
              5
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="6"
            >
              6
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="-"
            >
              -
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="1"
            >
              1
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="2"
            >
              2
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="3"
            >
              3
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="+"
            >
              +
            </button>
          </div>
          <div
            class="item zero"
          >
            <button
              type="button"
              value="0"
            >
              0
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="."
            >
              .
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="="
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components Home.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="p-5"
      >
        <h1>
          Welcome to our page!
        </h1>
        <br />
        <p>
          Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
        </p>
        <p>
          Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
        </p>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="p-5"
    >
      <h1>
        Welcome to our page!
      </h1>
      <br />
      <p>
        Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
      </p>
      <p>
        Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
      </p>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components Quote.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-center p-5 m-5"
      >
        <p
          class="w-50 fs-5 fw-boldtext-align-center"
        >
          Loading...
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-center p-5 m-5"
    >
      <p
        class="w-50 fs-5 fw-boldtext-align-center"
      >
        Loading...
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components testing Calculator.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex w-100 justify-content-evenly"
      >
        <div
          class="mt-5 w-50 px-5"
        >
          <h1>
            Let's do some Math's
          </h1>
        </div>
        <div
          class="d-flex justify-content-center w-50"
        >
          <div
            class="calculator mt-5"
          >
            <div
              class="result"
            >
              <span
                data-testid="result"
                id="result"
              >
                0
              </span>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="AC"
              >
                AC
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="+/-"
              >
                +/-
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="%"
              >
                %
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="รท"
              >
                รท
              </button>
            </div>
            <div
              class="item"
            >
              <button
                text="7"
                type="button"
                value="7"
              >
                7
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="8"
              >
                8
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="9"
              >
                9
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="x"
              >
                x
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="4"
              >
                4
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="5"
              >
                5
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="6"
              >
                6
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="-"
              >
                -
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="1"
              >
                1
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="2"
              >
                2
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="3"
              >
                3
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="+"
              >
                +
              </button>
            </div>
            <div
              class="item zero"
            >
              <button
                type="button"
                value="0"
              >
                0
              </button>
            </div>
            <div
              class="item"
            >
              <button
                type="button"
                value="."
              >
                .
              </button>
            </div>
            <div
              class="item operator"
            >
              <button
                type="button"
                value="="
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex w-100 justify-content-evenly"
    >
      <div
        class="mt-5 w-50 px-5"
      >
        <h1>
          Let's do some Math's
        </h1>
      </div>
      <div
        class="d-flex justify-content-center w-50"
      >
        <div
          class="calculator mt-5"
        >
          <div
            class="result"
          >
            <span
              data-testid="result"
              id="result"
            >
              0
            </span>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="AC"
            >
              AC
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="+/-"
            >
              +/-
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="%"
            >
              %
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="รท"
            >
              รท
            </button>
          </div>
          <div
            class="item"
          >
            <button
              text="7"
              type="button"
              value="7"
            >
              7
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="8"
            >
              8
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="9"
            >
              9
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="x"
            >
              x
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="4"
            >
              4
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="5"
            >
              5
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="6"
            >
              6
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="-"
            >
              -
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="1"
            >
              1
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="2"
            >
              2
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="3"
            >
              3
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="+"
            >
              +
            </button>
          </div>
          <div
            class="item zero"
          >
            <button
              type="button"
              value="0"
            >
              0
            </button>
          </div>
          <div
            class="item"
          >
            <button
              type="button"
              value="."
            >
              .
            </button>
          </div>
          <div
            class="item operator"
          >
            <button
              type="button"
              value="="
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components testing Home.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="p-5"
      >
        <h1>
          Welcome to our page!
        </h1>
        <br />
        <p>
          Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
        </p>
        <p>
          Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
        </p>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="p-5"
    >
      <h1>
        Welcome to our page!
      </h1>
      <br />
      <p>
        Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
      </p>
      <p>
        Curabitur aliquet quam id dui posuere blandit. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Nulla porttitor accumsan tincidunt. Donec sollicitudin molestie malesuada. Sed porttitor lectus nibh. Nulla porttitor accumsan tincidunt. Praesent sapien massa, convallis a pellentesque nec, egestas non nis. Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit. Donec rutrum congue leo eget malesuada.
      </p>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components testing Quote.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-center p-5 m-5"
      >
        <p
          class="w-50 fs-5 fw-boldtext-align-center"
        >
          Loading...
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-center p-5 m-5"
    >
      <p
        class="w-50 fs-5 fw-boldtext-align-center"
      >
        Loading...
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
